File: java/anagram/src/main/java/Anagram.java
import java.util.Arrays;
import java.util.List;
import java.util.stream.Collectors;

class Anagram {
    private String baseWord;
    private String nBaseWord;

    Anagram(String baseWord) {
        this.baseWord = baseWord;
        this.nBaseWord = normalize(baseWord);
    }

	public List<String> match(List<String> list) {
        return list.stream()
            .filter(s -> !s.toLowerCase().equals(baseWord.toLowerCase()))
            .filter(s -> normalize(s).equals(nBaseWord))
            .collect(Collectors.toList());
    }
    
    private String normalize(String string) {
        return Arrays.stream(string.toLowerCase().split(""))
            .sorted().collect(Collectors.joining());
    }
}

File: csharp/anagram/Anagram.cs
using System;
using System.Linq;

public class Anagram
{
    private string baseWord;

    public Anagram(string baseWord)
    {
        this.baseWord = baseWord;
    }

    public string[] Anagrams(string[] potentialMatches)
    {
        var nBaseWord = Normalize(baseWord);
        return potentialMatches
            .Where(w => w.ToLower() != baseWord.ToLower())
            .Where(w => Normalize(w) == nBaseWord)
            .ToArray();
    }

    private string Normalize(string w)
    {
        return string.Concat(w.ToLower().OrderBy(c => c));
    }
}